autonomic-algorithms-commons/src/main/java/br/com/autonomiccs/autonomic/algorithms/commons/resources/HostResources.java

```diff
@@ -28,7 +28,7 @@
 
 public class HostResources implements Cloneable {
 
-    private List<VmResources> vmsResources = new ArrayList<VmResources>();
+    private List<VmResources> vmsResources = new ArrayList<>();
     private float cpuOverprovisioning, memoryOverprovisioning;
```

Comment:
You are right. But, the point is that, we already use java 7, and we have to move to java 1.8. Oracle ceased public availability of security fixes and upgrades for Java 7 as of April 2015. We will not move backward.